### PR TITLE
docs(mesh) Add docs for mesh RBAC for new policies

### PR DESCRIPTION
**Pull request by Automaat**
Signed-off-by: Marcin Skalski [marcin.skalski@konghq.com](mailto:marcin.skalski@konghq.com)

Summary
Extend Kong Mesh docs with RBAC to include usage of new policies introduces in 2.0 version

Reason
https://github.com/Kong/kong-mesh/issues/1874

### CODE REVIEW

**Comment by Automaat** [comment]
@cloudjumpercat I rebased my changes from [PR](https://github.com/Kong/docs.konghq.com/pull/4664/files) with branch `release/mesh-2.0` and reopened this PR, but some checks are still failing. At least broken links check has passed now ;)

File: src/mesh/features/rbac.md [hunk 1]
```diff
@@ -71,6 +73,57 @@ rules:
 ```
 {% endnavtab %}
 {% endnavtabs %}
+{% endnavtab %}
+{% navtab `targetRef` selectors %}
+For policies using `targetRef` selector. You can specify which `targetRef` kinds users should have access to.
```
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:78]:
```suggestion
For policies using the `targetRef` selector. You can specify which `targetRef` kinds users should have access to.
```

File: src/mesh/features/rbac.md [hunk 2]
```diff
@@ -71,6 +73,57 @@ rules:
...
+{% navtabs %}
+{% navtab Kubernetes %}
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: AccessRole
+metadata:
+  name: role-1
+spec:
+  rules:
+  - types: ["MeshTrafficPermission", "MeshTrace", "MeshAccessLog"] # List of types that are granted access. If empty, then access is granted to all types.
```
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:89]:
```suggestion
  - types: ["MeshTrafficPermission", "MeshTrace", "MeshAccessLog"] # List of types that are granted access. If it's empty, access is granted to all types.
```

File: src/mesh/features/rbac.md [hunk 3]
```diff
@@ -71,6 +73,57 @@ rules:
...
+    names: ["res-1"] # List of allowed type names that are granted access. If empty, then access is granted to resources regardless of the name.
```
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:90]:
```suggestion
    names: ["res-1"] # List of allowed type names that are granted access. If it's empty, access is granted to resources regardless of the name.
```

File: src/mesh/features/rbac.md [hunk 4]
```diff
@@ -71,6 +73,57 @@ rules:
...
+        - name: k8s.kuma.io/namespace
+          value: kuma-demo
+```
+{% endnavtab %}
+{% navtab Universal %}
+```yaml
+type: AccessRole
+name: role-1
+rules:
+- types: ["MeshTrafficPermission", "MeshTrace", "MeshAccessLog"] # List of types that are granted access. If empty, then access is granted to all types.
```
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:109]:
```suggestion
- types: ["MeshTrafficPermission", "MeshTrace", "MeshAccessLog"] # List of types that are granted access. If it's empty, access is granted to all types.
```

File: src/mesh/features/rbac.md [hunk 5]
```diff
@@ -71,6 +73,57 @@ rules:
...
+  names: ["res-1"] # List of allowed type names that are granted access. If empty, then access is granted to resources regardless of the name.
```
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:110]:
```suggestion
  names: ["res-1"] # List of allowed type names that are granted access. If it's empty, access is granted to resources regardless of the name.
```

File: src/mesh/features/rbac.md [hunk 6]
```diff
@@ -226,6 +279,9 @@ This way a service owners can:
   It's useful because the service owner of backend has the best knowledge what (`Timeout`, `HealthCheck`) should be applied when communicating with their service.
 * Modify `TrafficTrace` or `ProxyTemplate` that matches backend service. This changes the configuration of data plane proxy that implements `backend` service.
 
+{:.note}
+> **Note**: When giving users `UPDATE` permission, remember to add `UPDATE` permission to all selectors they can switch between. For example, if a user only has access to `sources` selector, they won't be able to update policy with `destinations` selector or new `targetRef` selectors. Likewise, when a user only has access to `targetRef` kind `MeshService`, they won't be able to update the policy to use a different `targetRef` kind.
```
Comment (cloudjumpercat) [inline src/mesh/features/rbac.md:283]:
```suggestion
> **Note**: When giving users `UPDATE` permission, remember to add `UPDATE` permission to all selectors they can switch between. For example, if a user only has access to `sources` selector, they won't be able to update policy with `destinations` selector or new `targetRef` selectors. Likewise, when a user only has access to the `targetRef` kind `MeshService`, they won't be able to update the policy to use a different `targetRef` kind.
```

**Comment by cloudjumpercat** [comment]
@Automaat Michael just pushed a fix for the failing Rubocop tests to `release/mesh-2.0`, so those should be fixed now if you rebase. Sorry about that!

**Comment by lahabana** [comment]
@cloudjumpercat please merge this if you are happy with :)

**Pull request merged**